app/scripts/controllers/transactions/index.js

```diff
@@ -1471,24 +1471,17 @@ export default class TransactionController extends EventEmitter {
     const signedEthTx = await this.signEthTx(
       unsignedEthTx,
       fromAddress,
-      ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
-      txMeta.custodyStatus ? txMeta : undefined,
-      ///: END:ONLY_INCLUDE_IN
+      this._shouldDisablePublish(txMeta) ? txMeta : undefined,
```

Comment:
We're using this single callback in multiple places as the condition for all the existing MMI logic, but that means the controller still has to know the MMI specific logic but use a boolean callback to trigger it, so in essence what we have is a `isMMI` function we're using in multiple places.

Ideally what we want is completely decoupled logic injected through multiple callbacks.

This one for example could be an optional `getAdditionalSignArguments` callback as then we can support any arguments here rather than just the metadata which is what MMI wants.

Reply:
Created the `getAdditionalSignArguments` supporting any argument.